editor/src/clj/util/array.clj

```diff
@@ -0,0 +1,94 @@
+;; Copyright 2020-2024 The Defold Foundation
+;; Copyright 2014-2020 King
+;; Copyright 2009-2014 Ragnar Svensson, Christian Murray
+;; Licensed under the Defold License version 1.0 (the "License"); you may not use
+;; this file except in compliance with the License.
+;;
+;; You may obtain a copy of the License, together with FAQs at
+;; https://www.defold.com/license
+;;
+;; Unless required by applicable law or agreed to in writing, software distributed
+;; under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+;; CONDITIONS OF ANY KIND, either express or implied. See the License for the
+;; specific language governing permissions and limitations under the License.
+
+(ns util.array
+  (:import [java.lang.reflect Array]))
+
+(set! *warn-on-reflection* true)
+(set! *unchecked-math* :warn-on-boxed)
+
+(def ^"[Ljava.lang.Object;" empty-object-array (Array/newInstance Object 0))
+
+(defn of
```

Comment:
Intended use: `(array/of "item one" :item-two ItemThree)`.

Perhaps we'll add primitive flavors like `of-longs`, `of-doubles` and so on in the future.

I decided to make this its own module so I could use it from `util.eduction` as well.